src/block-components/helpers/backgrounds/style.js

```diff
@@ -41,9 +41,14 @@ const getStyleParams = ( options = {} ) => {
 					getAttribute( 'backgroundMediaUrl', 'mobile' )
 
 				if ( ! backgroundColorType && backgroundColorOpacity !== '' && ! hasBackground ) {
+					// Checks if color comes from Blocksy color pallete.
+					if ( value?.startsWith( 'var(--paletteColor' ) ) {
+						const hexColorIndex = value.search( /#[a-zA-Z0-9]{6}/g )
+						const hexColor = value.substring( hexColorIndex, hexColorIndex + 7 )
```

Comment:
If you check:
- `src/block-components/typography/edit.js`
- `src/block-components/button/edit.js`

they have this line:
```js
const value = _value?.startsWith( 'var(--stk-global-color' ) ? _value.match( /(#[^\)]*)/g )[ 0 ] : _value
```
which gets the color from a CSS custom property, similar to what you're doing.

I refactored the extracting code to support both `--stk-global-color` and `--paletteColor` so we can reuse it in your PR and in the 2 files I listed above. Add a new util function in `src/util` and use it in the codes mentioned (and in your fix):
```js
export const extractColor = value => {
	if ( color?.startsWith( 'var(--' ) ) {
		if ( color.match( /(#[^\)]*)/g ) ) {
			return color.match( /(#[^\)]*)/g )[ 0 ]
		}
	}
	return color
}

Reply:
Got it